services/hanke-service/src/main/resources/db/changelog/changesets/018-add-haittojen-hallinta-fields.yml

```diff
@@ -0,0 +1,72 @@
+databaseChangeLog:
+  - changeSet:
+      id: 018-add-haittojen-hallinta-fields
+      comment: Add fields for managing haittojen hallinta
+      author: Niko Junkala
+      changes:
+        - createTable:
+            tableName: haittojenhallinta_kuvaukset
+            remarks: "Stores haittojen hallinta kuvaukset as key-value pairs"
+            columns:
+            - column:
+                name: id
+                type: bigint
+                constraints:
+                  primaryKey: true
+            - column:
+                name: haittojenhallinta_id
+                type: bigint
+                constraints:
+                  nullable: false
+                  unique: true
+            - column:
+                name: key
+                remarks: "Haittojen hallinta description key. Uses values of enum HaittojenHallintaKentta."
+                type: varchar(64) # enum HaittojenHallintaKentta
+                constraints:
+                  nullable: false
+            - column:
+                name: value
+                remarks: "Haittojen hallinta description. Free text."
+                type: text
+
+        - createTable:
+            tableName: haittojenhallinta
+            remarks: "Stores haittojen hallinta. This is to be used with Hanke and Hakemus."
+            columns:
+              - column:
+                  name: id
+                  type: bigint
+                  constraints:
+                    primaryKey: true
+              - column:
+                  name: kuvaukset_id
+                  remarks: "References descriptions"
+                  type: bigint
+                  constraints:
+                    referencedTableName: "haittojenhallinta_kuvakset"
+                    foreignKeyName: "haittojenhallinta_id"
```

Comment:
Eikö tämä viiteavain ole väärään suuntaan? `haittojenhallinta_kuvaukset` viittaa tähän tauluun, eikä toisin päin. Tätä ei taida olla entityissä ollenkaan.

Johtuukohan tuosta typosta referencedTableNamessa, että tota viiteavain-rajoitetta ei luoda ollenkaan. Vähän pelottavaa, jos Liquibase tolleen vaan hiljaisesti ohittaa virheelliset määrittelyt.

Reply:
Tämä kommentti oli suljettu, mutta mikään ei ole muuttunut?